### PR TITLE
[Core] Fix issues with MinGW compilation

### PR DESCRIPTION
**Pull request by loumalouomega**
Just that

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -141,28 +141,40 @@ if(${CMAKE_COMPILER_IS_GNUCXX})
...
+  if(${CMAKE_SYSTEM_NAME} MATCHES "Windows")
+  	  if(${CMAKE_CXX_COMPILER_ID} MATCHES Clang)
+		set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}  -funroll-loops -Wall -Wno-unused-local-typedef -Wno-unknown-pragmas  ")
+		set(CMAKE_C_FLAGS   "${CMAKE_C_FLAGS} -funroll-loops -Wall -Wno-unknown-pragmas  ")
+		set(CMAKE_SHARED_LINKER_FLAGS "-Wl,--no-undefined")
+		message("additional default options were set for clang compiler")
+		message("CMAKE_CXX_FLAGS = ${CMAKE_CXX_FLAGS}")
+		message("CMAKE_C_FLAGS = ${CMAKE_C_FLAGS}")
+	  endif(${CMAKE_CXX_COMPILER_ID} MATCHES Clang)
+  else(${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
```
Comment (roigcarlo) [inline CMakeLists.txt]: I am pretty sure it will never never enter here. (Lines 146 + 155 = False)

File: scripts/standard_configure_MINGW.sh [hunk 2]
```diff
@@ -40,6 +40,7 @@ add_app ${KRATOS_APP_DIR}/ConvectionDiffusionApplication
 cmake ..                                                                                            \
 -G "MinGW Makefiles"                                                                                \
 -DWIN32=TRUE                                                                                        \
+-DCMAKE_INSTALL_PREFIX="${KRATOS_SOURCE}/bin/${KRATOS_BUILD_TYPE}"                                  \
```
Comment (roigcarlo) [inline scripts/standard_configure_MINGW.sh:43]: This variable with this value should be automatically set by CMake

File: CMakeLists.txt (hunk 1 above)
Comment (loumalouomega) [reply CMakeLists.txt]: Sorry, copy paste

File: scripts/standard_configure_MINGW.sh (hunk 2 above)
Comment (loumalouomega) [reply scripts/standard_configure_MINGW.sh:43]: For some reason it is not

Comment (roigcarlo) [reply scripts/standard_configure_MINGW.sh:43]: :+1: I'll take a look

**Review by roigcarlo — APPROVED** [review]

**Pull request merged**